### ISSUE TITLE
Support Android Development from 32-bit Windows

### ISSUE THREAD

**Issue by eseidelGoogle** [flutter/flutter#33862]
At this time we have no plans to add 32-bit support for Windows. I'm filing this bug as a cleaner place to track this request (e.g. upvote if you want this) as well as the blocking issues were someone to want to add support in the future.

We have several other existing 32-bit windows bugs, all of which are closed, e.g. https://github.com/flutter/flutter/issues/22638 https://github.com/flutter/flutter/issues/21169 https://github.com/flutter/flutter/issues/14925

As originally captured in https://github.com/flutter/flutter/issues/14925#issuecomment-415864259, blockers are at least:
1. flutter/engine has no 32-bit build rules currently, including gen_snapshot
2. gen_snapshot is architecture dependent, unclear if a 32-bit gen_snapshot could build for 64-bit Android (which is a requirement for the Play Store as of this August).

Steam stats show 32-bit Windows as less than 2% market share:
https://store.steampowered.com/hwsurvey

Given the low market share numbers and [other vendors dropping support for 32-bit](https://www.extremetech.com/computing/267180-nvidia-ends-support-for-32-bit-operating-systems), it seems highly unlikely we'd add 32-bit Windows development support for Flutter.

I welcome thoughts from others, or additional/new data as to why we should add 32-bit development support?

**Comment by duzenko** [comment]
I'm on x64 and I still want the 32-bit dart sdk for one simple reason: the background dart.exe processed Android Studio spawns waste too much RAM
![image](https://user-images.githubusercontent.com/1481807/60422919-b0fa5780-9bf5-11e9-955b-0ca104378aeb.png)

**Comment by anthonyr013** [comment]
better use nativescript

**Comment by Sunbreak** [comment]
@eseidelGoogle Maybe we can close the issure.

Google deprecated 32-bit Windows support in Android Studio since December 31, 2019
https://android-developers.googleblog.com/2019/06/moving-android-studio-and-android.html

**Comment by duzenko** [comment]
> 
> 
> @eseidelGoogle Maybe we can close the issure.
> 
> Google deprecated 32-bit Windows support in Android Studio since December 31, 2019
> https://android-developers.googleblog.com/2019/06/moving-android-studio-and-android.html

I don't see how this is relevant to anything unless you think Flutter = Android

**Comment by Sunbreak** [comment]
@duzenko The title is `Android Development`

**Comment by duzenko** [comment]
> 
> 
> @duzenko The title is `Android Development`

See all the issues closed linking to this one

**Comment by Sunbreak** [comment]
BTW I'm working on `Flutter for Windows` and mainly targeting at Windows x64
Maybe target at Windows x86 is an option. But I don't think develop on Windows x86 is neccessary

**Comment by Hari-07** [comment]
i saw in #14925 that you'd gladly review PRs if someone wants to work on extending it to work on 32 bit systems. Has there been any work done in that direction since? I'm interested in trying to contribute to that

**Comment by Sunbreak** [comment]
> i saw in #14925 that you'd gladly review PRs if someone wants to work on extending it to work on 32 bit systems. Has there been any work done in that direction since? I'm interested in trying to contribute to that

@Hari-07 @duzenko 

Google deprecated 32-bit Windows support in Android Studio since December 31, 2019: https://android-developers.googleblog.com/2019/06/moving-android-studio-and-android.html. Which means you're not able to use Android Studio on Windows x86 afterwards

It is suggested to get a Windows x64 PC as develop environment

If you prefer Windows x86 as a **target**, thumb up #37777 please

**Comment by Hari-07** [comment]
I do use flutter for app dev on my x64 PC , however i think it might be worthwhile to get this working as places like schools who have old systems might benefit. Android studio, i believe older versions would still work on 32 bit systems just as windows 7 has been deprecated but still is used , and 32 bit win7 is almost twice as popular as 32 bit win 10. and as @duzenko said flutter can or not support it , independent of android studio's decision.

I am not experienced with anything related to this, but i think this'd be a good project atm to try and get into.

**Comment by duzenko** [comment]
Android Studio being Java application runs with no problem in the 32-bit environment
But the Flutter team seems to be regretfully overstretched and barely able to keep up with essentials. Support for 32 bit dart SDK is unlikely to happen for the simplest of reason - deficit of the support manpower.

**Comment by Sunbreak** [comment]
> Android Studio being Java application runs with no problem in the 32-bit environment
> But the Flutter team seems to be regretfully overstretched and barely able to keep up with essentials. Support for 32 bit dart SDK is unlikely to happen for the simplest of reason - deficit of the support manpower. Everything else is merely excuses.

Agreed. It is possible to support Windows x86 as develop environment

But is it worth the effort?

Deprecation of Windows x86 support in Android Studio **suggests** it is time to move on

**Comment by Sunbreak** [comment]
Global: Win10 (53.74%) vs Win7 (28.35%)
https://netmarketshare.com/operating-system-market-share.aspx?id=platformsDesktopVersions

China: Win10 (47.14%) vs Win7 (46.79%)
https://gs.statcounter.com/windows-version-market-share/desktop/china

As a user I'd like to use APPs on Win7(mostly x86). But as a developer I'm required to keep up with the newest environment

**Comment by m2dr00** [comment]
hi you are just like other evil company you say black lives matter in golang website But what about us? My only guilt is that I was born in Iran and because in Iran I do not have the right to use the services of Google or Amazon or Facebook like the rest of the world, a 64-bit computer is nothing for you, but for me who is Iranian, a 64-bit laptop size 4 My salary is expensive, and if I want to buy a laptop, I must not eat anything for 4 months, and this means that my death is a curse on Google and human rights. If I am not a human being, then why should I not have a right to my world?
google just a Lier like the other shame on you.

Rest assured that the day I die, I will complain to you before God and know that God is the greatest and most just judge in the world. It's me and my compatriots. I can not breathe because you only pretend to be human. I do not see any human among you. I am Iranian and that means I am Persian and not Arab. I can not breathe because your foot is on my neck. 

Please let me breathe. Lift the embargo on Iranian developers. I would like to go to Go Go site without a vpn. I would like to use Android Studio, which is free, without a vpn. I love coding, but you have taken this right from me. At least Do not deprive us Iranian programmers of the right to code with a filter. And publish a 32-bit flutter. God exists and you only oppress. I only have a salary of $ 50 in Iran and I live on it every month. You can help me $ 10 per person so that I can buy a laptop for the whole money. I need $ 350 and ten dollars is too little for you, not for me. I do not eat meat every year because I can not buy I can not have a Samsung A5 mobile phone because the total salary is one month. If you want to help me, send me an email I leave this comment at the bottom (m2dr000@gmail.com). Email me to send you proof of proof, then see my life, and if you did not cry or be upset, know that the person inside you is dead. I only eat one meal a day, some bread and mayonnaise. What do you do during Do you eat during the day? I can have some of your food, just a small piece of pizza. I have been wanting to eat pizza once in the 20 years I have lived. You are very lucky. I cried when I saw the fat or muscular bodies of Europeans and Americans because I am a bunch of bones. I see in your movies that they break eggs or throw them and I just cry why. I can not eat a grain of it and they have so much that they throw them. I only believe in God because in this world I only suffered and cursed my life.
My email: m2dr000@gmail.com
And my number:
+989158225066

**Comment by Sunbreak** [comment]
@m2dr00 Quite sorry about that

Could we build 64-bit APPs on 32-bit computor?
AFAIK, both Google Play Store & Apple App Store require APPs built with 64-bit, whether or not 32-bit supported

**Comment by duzenko** [comment]
> 
> 
> Could we build 64-bit APPs on 32-bit computor?
> AFAIK, both Google Play Store & Apple App Store require APPs built with 64-bit, whether or not 32-bit supported

What is one single reason for not being able to cross compile for ARM64 (or any other architecture for that matter) on a 32-bit computer (other than build crappy tools requiring too much RAM for no reason)

**Comment by Piero512** [comment]
I wonder if there are enough Windows on ARM laptops that might benefit from being able to work with x86 emulation, since MS hasn't (at the time of writing) emulated x86_64 apps on ARM. But then, those devices are really slow.

**Comment by Sunbreak** [comment]
> I wonder if there are enough Windows on ARM laptops that might benefit from being able to work with x86 emulation, since MS hasn't (at the time of writing) emulated x86_64 apps on ARM. But then, those devices are really slow.

@Piero512 https://blogs.windows.com/windows-insider/2020/12/10/introducing-x64-emulation-in-preview-for-windows-10-on-arm-pcs-to-the-windows-insider-program/

**Comment by xyun52** [comment]
百度统计 操作系统排名
https://tongji.baidu.com/research/site?source=index#screen
![微信截图_20210831133948](https://user-images.githubusercontent.com/5417624/131448325-be3cf0ba-98bd-4c15-bc3f-2863b9d2740f.png)

**Comment by Sunbreak** [comment]
> 百度统计 操作系统排名
> https://tongji.baidu.com/research/site?source=index#screen
> ![微信截图_20210831133948](https://user-images.githubusercontent.com/5417624/131448325-be3cf0ba-98bd-4c15-bc3f-2863b9d2740f.png)

@xyun52 标题是“在32位Windows上支持Flutter Android“开发，你应该想要的是支持Flutter Windows支持32位吧：https://github.com/flutter/flutter/issues/37777

**Comment by harounakinda** [comment]
We are now in 2022, so the 32-bit flutter sdk is still not available?

**Comment by shichaoxxx** [comment]
都2023年了 还不支持吗

**Comment by Sunbreak** [comment]
> 都2023年了 还不支持吗

这个issue是“在32位Windows上开发Flutter”，不是把“32位Windows作为目标产物”

**Comment by Hari-07** [comment]
VSCode stopped supporting x86 sometime in 2023, we can probably close this as not planned. 
@jmagman
